Backend/Tumultu/Application/FileVariants/Queries/GetFileVariantsByFileQuery.cs

```diff
@@ -0,0 +1,28 @@
+﻿using Ardalis.GuardClauses;
+using MediatR;
+using Tumultu.Application.Common.Interfaces;
+using Tumultu.Domain.Entities;
+
+namespace Tumultu.Application.FileVariants.Queries;
+
+public record GetFileVariantsByFileQuery : IRequest<IEnumerable<FileVariant>>
+{
+    public FileEntity? File { get; init; }
+}
+
+public record GetFileVariantsByFileQueryHandler : IRequestHandler<GetFileVariantsByFileQuery, IEnumerable<FileVariant>>
+{
+    private readonly IReadRepository<FileVariant, Guid> _repository;
+    public GetFileVariantsByFileQueryHandler(IReadRepository<FileVariant, Guid> repository)
+    {
+        _repository = repository;
+    }
+
+    public async Task<IEnumerable<FileVariant>> Handle(GetFileVariantsByFileQuery request, CancellationToken cancellationToken)
+    {
+        IEnumerable<FileVariant> entities = (await _repository.GetAllAsync())
```

Comment:
The filtering should be implemented on SQL level since it is a Dapper implementation, otherwise it will actually pull all records from database and filter them in code.